Create storage

RUMirNul opened the pull request.

Commit:
Add BaseIdEntity
Add UserEntity
Add BookEntity

Commit:
Add UserDao
Add UserDaoImpl
Add BookDao
Add BookDaoImpl

Commit:
Update UserMapper
Update BookMapper

Commit:
Add BookRepository
Add BookRepositoryImpl
Add BookStorage
Add UserRepository
Add UserRepositoryImpl
Add UserStorage
Update Storage

Commit:
Update BookService
Update BookServiceImpl
Update UserServiceImpl

Commit:
Update UserController
Update UserDataFacade

Commit:
Update ControllerExceptionHandler
Add ErrorMessageTextConstants

RUMirNul committed:
Create README.md

Add readme.md

Lada303 commented on src/main/java/com/edu/ulab/app/dao/BookDao.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.util.List;
+
+public interface BookDao<E, K> {
```
К наверно излишне, все же мы знаем тип id

Lada303 commented on src/main/java/com/edu/ulab/app/entity/BookEntity.java:
```diff
@@ -0,0 +1,27 @@
+
+@Data
+@EqualsAndHashCode(callSuper = false)
+@Builder
```
тут можно еще аннотации по конструкторам подвесить, и еще кое-какие...

Lada303 commented on src/main/java/com/edu/ulab/app/storage/impl/Storage.java:
```diff
@@ -0,0 +1,176 @@
+@Repository
+@Slf4j
+public class Storage implements UserStorage<UserEntity, Long>, BookStorage<BookEntity, Long> {
+
```
Имплементацию тоже надо было делать два разных класса.

Lada303 commented on src/main/java/com/edu/ulab/app/storage/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,105 @@
+@Repository
+@Slf4j
+public class BookRepositoryImpl implements BookRepository<BookEntity, Long> {
+
```
репозитории и сторадж - одно и тоже по смыслу - зачем и то и другое? По-моему сильно раздробил слои
Также все эти проверки на нулл и т.п. -- должны производиться в сервис слои или фасаде. Тут только работа с хранилищем и все.

Lada303 commented on src/main/java/com/edu/ulab/app/web/handler/ControllerExceptionHandler.java:
```diff
@@ -14,8 +14,8 @@
 
-    @ExceptionHandler(NotFoundException.class)
-    public ResponseEntity<BaseWebResponse> handleNotFoundExceptionException(@NonNull final NotFoundException exc) {
+    @ExceptionHandler({NotFoundException.class, NullPointerException.class})
```
тут все же предполагалось что будет кастомные ошибки, несколько методов перехвата, каждый со своим уникальным сообщением и красивым ответом на фронт (с разными HttpStatus и т.п.)

Lada303 commented on src/main/java/com/edu/ulab/app/service/impl/BookServiceImpl.java:
```diff
@@ -1,31 +1,49 @@
+        this.bookDao = bookDao;
+        this.bookMapper = bookMapper;
+    }
+
```
а тут нет логирования, хотя аннотация @Slf4j подключена

Commit:
Created InvalidRequestDataException
Created handler for InvalidRequestDataException
Removed throwing errors from the repository layer
Updated the logic of throwing errors

Commit:
deleted ignore readme.md

Commit:
Changed annotations

Commit:
Added logs for BookServiceImpl
Added logs for UserServiceImpl
